waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -337,3 +337,50 @@
                     :interval 15
                     :timeout timeout-secs)))))
         (log/warn "skipping test as INTEGRATION_TEST_BAD_IMAGE is not specified")))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar-headers
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (when-let [proxy-options (get-k8s-proxy-options waiter-url)]
+        (let [{:keys [service-id] :as response}
+              (make-request-with-debug-info
+                {:x-waiter-distribution-scheme "simple"
+                 :x-waiter-name (rand-name)
+                 (keyword (str "x-waiter-env-" (:reverse-proxy-flag proxy-options))) "yes"}
+                #(make-kitchen-request waiter-url % :method :get :path "/request-info"))]
+          (with-service-cleanup
+            service-id
+            (assert-response-status response http-200-ok)
+            (testing "Expected envoy specific headers are present in both request and response"
+              (let [response-body (try-parse-json (:body response))
+                    response-headers (:headers response)]
+                (is (contains? (get response-body "headers") "x-envoy-expected-rq-timeout-ms"))
+                (is (contains? response-headers "x-envoy-upstream-service-time"))
+                (is (= "envoy" (get response-headers "server")))))))))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar-env
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (when-let [{:keys [reverse-proxy-flag reverse-proxy-offset]} (get-k8s-proxy-options waiter-url)]
+        (let [{:keys [service-id] :as response}
+              (make-request-with-debug-info
+                {:x-waiter-distribution-scheme "simple"
```

Comment:
Why do we need simple distribution for this test?

Reply:
Resolved above